src/pykeen/training/training_loop.py

```diff
@@ -636,8 +636,10 @@ def _forward_pass(self, batch, start, stop, current_batch_size, label_smoothing,
         loss.backward()
         current_epoch_loss = loss.item()
 
-        # reset the regularizer to free the computational graph
-        self.model.regularizer.reset()
+        if hasattr(self.model, 'regularizer'):
+            # reset the regularizer to free the computational graph
+            self.model.regularizer.reset()
+            # TODO why not call torch.cuda.empty_cache()? or call self._free_graph_and_cache()?
```

Comment:
@lvermue I think you wrote the `_free_graph_and_cache()` function. Could you answer this question - why is `torch.cuda.empty_cache()` not called after resetting the regularizer?